ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/services/session-storage.ts

```diff
@@ -6,7 +6,7 @@ export class SessionStorage<TType> {
      */
     get(): TType {
         const cached = sessionStorage.getItem(this.storageKey);
-        if (!cached) {
+        if (!cached || cached === undefined || cached === 'undefined') {
```

Comment:
this doesn't quite look right. we shouldn't allow to store it as 'undefined' it must be a bug somewhere. maybe better check in the `set()` method that we don't allow `null` or `undefined` to be set?

then `!cached` should be enough to raise if it's not set.

Reply:
I see your point, but we can't control everyone from not having a bug and storing undefined. As we can reliably say that no one will ever WANt to store Undefined, we can defend here for occasions where it is stored... What you think ?

Reply:
Changed to check for undefined when trying to set the cached value